editor/editor_file_dialog.cpp

```diff
@@ -121,6 +121,11 @@ void EditorFileDialog::_notification(int p_what) {
 		if (!is_visible()) {
 			set_process_unhandled_input(false);
 		}
+	} else if (p_what == NOTIFICATION_WM_WINDOW_FOCUS_IN) {
+		// Check if the current directory was removed externally (much less likely to happen while editor window is focused).
+		while (!dir_access->dir_exists(get_current_dir())) {
+			_go_up();
+		}
```

Comment:
I wonder if this would become an infinite loop. e.g. You were somewhere in Windows `E:` drive, but later the drive is  removed?

Reply:
I though about that and didn't considered that a drive can indeed be removed. The problems is where to go when this happens?

Reply:
I guess on Windows we can expect `C:\` to always be available? Or `%USERPROFILE%`?

Reply:
The drives available can be retrieved with `get_drive_count()` and `get_drive(n)`. Maybe just try them one by one.

Reply:
Added fallback to user directoy. I can't test if it works though, I don't have external drive and deleting project directory with project open is impossible on Windows.